File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/character/choice/CharacterBaseInformations.scala





















// Generated on 05/03/2013 10:35:03
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait CharacterBaseInformations extends CharacterMinimalPlusLookInformations {
  override val typeId = 45

  protected trait Type extends CharacterMinimalPlusLookInformations.Type with Serializable {
    override val typeId = CharacterBaseInformations.typeId

    val breed: Byte
    val sex: Boolean

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeByte(breed)
      buf.writeBoolean(sex)
    }

  }

  protected def create(id: Int, level: UByte, name: String, entityLook: EntityLook.TypeImpl, breed: Byte, sex: Boolean): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val id = buf.readInt()
    val level = buf.readUByte()
    val name = buf.readString()
    val entityLook = EntityLook.deserialize(buf)
    val breed = buf.readByte()
    val sex = buf.readBoolean()

    create(id, level, name, entityLook, breed, sex)
  }

}

object CharacterBaseInformations extends CharacterBaseInformations {
  case class TypeImpl(id: Int, level: UByte, name: String, entityLook: EntityLook.TypeImpl, breed: Byte, sex: Boolean) extends Type
  override def create(id: Int, level: UByte, name: String, entityLook: EntityLook.TypeImpl, breed: Byte, sex: Boolean) = TypeImpl(id, level, name, entityLook, breed, sex)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(id: Int, level: UByte, name: String, entityLook: EntityLook.TypeImpl, breed: Byte, sex: Boolean) = create(id, level, name, entityLook, breed, sex)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/CharacterBaseInformations.java


// Generated on 05/08/2013 19:38:00
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class CharacterBaseInformations extends CharacterMinimalPlusLookInformations {
    public static final short TYPE_ID = 45;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public byte breed;
    public boolean sex;
    
    public CharacterBaseInformations() { }
    
    public CharacterBaseInformations(int id, short level, String name, EntityLook entityLook, byte breed, boolean sex) {
        super(id, level, name, entityLook);
        this.breed = breed;
        this.sex = sex;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeByte(breed);
        buf.writeBoolean(sex);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        breed = buf.readByte();
        sex = buf.readBoolean();
    }
    
}
